lib/Platform/Intl/BCP47Parser.cpp

```diff
@@ -73,43 +73,50 @@ bool isUnicodeLanguageSubtag(const std::u16string &str, int start, int end) {
   // https://unicode.org/reports/tr35/#Unicode_Language_and_Locale_Identifiers
   // = alpha{2,3} | alpha{5,8};
   // root case?
-  return isCharType(str, start, end, 2, 3, &isASCIILetter) ||
-    isCharType(str, start, end, 5, 8, &isASCIILetter);
+  const llvh::ArrayRef<char16_t> ref(str.data() + start, end - start + 1);
+  return isCharType(ref, 2, 3, &isASCIILetter) ||
+    isCharType(ref, 5, 8, &isASCIILetter);
 }
 bool isUnicodeScriptSubtag(const std::u16string &str, int start, int end) {
```

Comment:
All of these functions should now take `ArrayRef`, instead of needing to take the underlying string and a range.